Common/src/matrix_structure.cpp

```diff
@@ -1040,13 +1142,45 @@ void CSysMatrix::InverseDiagonalBlock(unsigned long block_i, su2double *invBlock
       invBlock[jVar*nVar+iVar] = aux_vector[jVar];
   }
   
+  //  su2double Det, **Matrix, **CoFactor;
```

Comment:
That is an alternative method to compute the inverse of a block. I have had a lot os problems with the ILU0 preconditioned for grid deformation... in ~75% of my problems it doesn't work... and Discrete adjoint doesn't work either. I thought the problem was in some kind of singular matrix, that is the reason I implement some alternatives to existing subroutines. The commented one is more elegant, and we will probably use that in the future. I'll create a new branch to work on the problem.